data/test/scenarios/backstab.cfg

```diff
@@ -82,14 +82,80 @@
 #enddef
 
 # The test uses a Rogue to do the attack, which does 6x3 damage before the ability is applied
+
+#####
+# API(s) being tested: [damage]+[filter_opponent]formula
+##
+# Starting state:
+# Side 1 leader Alice (Elvish Archer)
+# Side 2 leader Bob (Orcish Grunt)
+##
+# Actions:
+# Bob is put on Void terrain (0% defense) and given 100 hp.
+# A Rogue is spawned north and south of Bob.
+# A Rogue attacks Bob.
+##
+# Expected end state:
+# Bob gets hit all three times and ends with 100-(6 * 3 * 2) hp.
```

Comment:
I feel this adds a lot of lines of text, but the only new information added is that it's testing `[damage]+[filter_opponent]formula`; and it requires a lot more reading to compare the differences between tests. It also means that checking for missing test cases requires scanning a lot more vertical space.

For understanding and maintaining this set of tests, I think the common docs in lines 4-9 are more useful, and I feel the GENERIC_UNIT_TEST should probably be documented in a single central location.

There are a couple of ways to force 100% chance to hit, of which using void terrain is merely one of them. Putting the exact method in the docs here means more changes if someone refactors the mechanism.

Reply:
That was the point - each test should be documented as to what it's doing individually.

Reply:
But what advantages do we gain from that, compared to the disadvantages above?

Reply:
I think the parts about GENERIC_UNIT_TEST and the method of making the unit have 0% defense are true.  I don't feel that lines 4-9 fulfill the same purpose as the comment blocks I've added though.

Reply:
Instead of documenting the mechanics of the test, how about changing two of those lines to:
* "Bob is made to have 0% defense but given enough HP to survive."
* "The backstab ability was active, as shown by the amount of damage that Bob takes."

Reply:
> I guess that logic implies that these lines should just have `yes` or `no` instead of `(6 * 3 * 2)`.

I've created #6473 with that change to the logic, and started with the version of the comments from this PR.